Tests/Fluent.Tests/Compute/SharedGalleryImageTests.cs

```diff
@@ -0,0 +1,337 @@
+
+                    Assert.NotNull(galleryImage);
+                    Assert.NotNull(galleryImage.Inner);
+                    //
+                    // Delete an image from gallery
+                    //
+                    azure.GalleryImages.DeleteByGallery(rgName, galleryName, galleryImageName);
+                }
+                finally
+                {
+                    try
+                    {
+                        azure.ResourceGroups.DeleteByName(rgName);
+                    }
+                    catch { }
+                }
+            }
+        }
+
+        [Fact]
+        public void CanCreateUpdateGetDeleteGalleryImageVersion()
+        {
+            using (var context = FluentMockContext.Start(GetType().FullName))
+            {
+                Region region = Region.USWestCentral;
+                string rgName = TestUtilities.GenerateName("vmexttest");
+                string galleryName = TestUtilities.GenerateName("jsim");
+                string galleryImageName = "SampleImages";
+
+                var azure = TestHelper.CreateRollupClient();
+
+                try
+                {
+
+                    IGallery gallery = azure.Galleries.Define(galleryName)
+                            .WithRegion(region)
+                            .WithNewResourceGroup(rgName)
+                            .WithDescription("java's image gallery")
+                            .Create();
+                    //
+                    // Create an image in the gallery (a container to hold custom linux image)
+                    //
+
+                    IGalleryImage galleryImage = azure.GalleryImages.Define(galleryImageName)
+                            .WithExistingGallery(gallery)
+                            .WithLocation(region)
+                            .WithIdentifier("JavaSDKTeam", "JDK", "Jdk-9")
+                            .WithGeneralizedLinux()
+                            .Create();
+                    //
+                    // Create a custom image to base the version on
+                    //
+                    IVirtualMachineCustomImage customImage = PrepareCustomImage(rgName, region, azure);
+                    // String customImageId = "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/javacsmrg91482/providers/Microsoft.Compute/images/img96429090dee3";
+                    //
+                    // Create a image version based on the custom image
+                    //
+
+                    string versionName = "0.0.4";
+
+                    IGalleryImageVersion imageVersion = azure.GalleryImageVersions.Define(versionName)
+                            .WithExistingImage(rgName, gallery.Name, galleryImage.Name)
+                            .WithLocation(region.ToString())
+                            .WithSourceCustomImage(customImage)
+                            // Options - Start
+                            .WithAvailableInRegion(Region.USEast2)
```

Comment:
>WithAvailableInRegion [](start = 29, length = 21)

`WithRegionAvailability` ?

Reply:
thinking more about it.. we also have a wither that takes multiple regions `WithAvailableInRegions(List<Region>)`. For consistency if we replace it with `WithRegionsAvailability(List<Region>)`, not sure how readable it is.

Reply:
ok, discussed with Asir, he recommended having overloads of `WithAvailableInRegion` that takes single region and list. Updated the PR.